lib/api/src/modules/layout.ts

```diff
@@ -1,67 +1,135 @@
+}
+
+export interface SubAPI {
+  toggleFullscreen: (toggled?: boolean) => void;
+  togglePanel: (toggled?: boolean) => void;
+  togglePanelPosition: (position?: PanelPositions) => void;
+  toggleNav: (toggled?: boolean) => void;
+  toggleToolbar: (toggled?: boolean) => void;
+  setOptions: (options: any) => void;
+}
+
+type PartialSubState = Partial<SubState>;
+type PartialThemeVars = Partial<ThemeVars>;
+type PartialLayout = Partial<Layout>;
+type PartialUI = Partial<UI>;
+
+interface Options {
+  name?: string;
+  url?: string;
+  goFullScreen: boolean;
+  showStoriesPanel: boolean;
+  showAddonPanel: boolean;
+  addonPanelInRight: boolean;
+  theme?: ThemeVars;
+  selectedPanel?: string;
+}
+
+interface OptionsMap {
+  [key: string]: string;
+}
 
-const deprecatedThemeOptions = {
+const deprecatedThemeOptions: {
+  name: 'brandTitle';
+  url: 'brandUrl';
+} = {
   name: 'brandTitle',
   url: 'brandUrl',
 };
 
-const deprecatedLayoutOptions = {
+const deprecatedLayoutOptions: {
+  goFullScreen: 'isFullscreen';
+  showStoriesPanel: 'showNav';
+  showAddonPanel: 'showPanel';
+  addonPanelInRight: 'panelPosition';
+} = {
   goFullScreen: 'isFullscreen',
   showStoriesPanel: 'showNav',
   showAddonPanel: 'showPanel',
   addonPanelInRight: 'panelPosition',
 };
 
-const deprecationMessage = (optionsMap, prefix) =>
+const deprecationMessage = (optionsMap: OptionsMap, prefix: string = '') =>
   `The options { ${Object.keys(optionsMap).join(', ')} } are deprecated -- use ${
     prefix ? `${prefix}'s` : ''
   } { ${Object.values(optionsMap).join(', ')} } instead.`;
 
 const applyDeprecatedThemeOptions = deprecate(
-  ({ name, url }) => ({
+  ({ name, url }: Options): PartialThemeVars => ({
     brandTitle: name,
     brandUrl: url,
     brandImage: null,
   }),
   deprecationMessage(deprecatedThemeOptions)
 );
 
-const applyDeprecatedLayoutOptions = deprecate(options => {
-  const layoutUpdate = {};
+const applyDeprecatedLayoutOptions = deprecate((options: Options): PartialLayout => {
+  const layoutUpdate: PartialLayout = {};
 
-  ['goFullScreen', 'showStoriesPanel', 'showAddonPanel'].forEach(option => {
-    if (typeof options[option] !== 'undefined') {
-      layoutUpdate[deprecatedLayoutOptions[option]] = options[option];
+  ['goFullScreen', 'showStoriesPanel', 'showAddonPanel'].forEach(
+    (option: 'goFullScreen' | 'showStoriesPanel' | 'showAddonPanel') => {
+      const v = options[option];
+      if (typeof v !== 'undefined') {
+        const key = deprecatedLayoutOptions[option];
+        layoutUpdate[key] = v;
+      }
     }
-  });
+  );
   if (options.addonPanelInRight) {
     layoutUpdate.panelPosition = 'right';
   }
   return layoutUpdate;
 }, deprecationMessage(deprecatedLayoutOptions));
 
-const checkDeprecatedThemeOptions = options => {
-  if (Object.keys(deprecatedThemeOptions).find(key => !!options[key])) {
+const checkDeprecatedThemeOptions = (options: Options) => {
+  if (Object.values(deprecatedThemeOptions).find(v => !!v)) {
```

Comment:
What is this check intended to do? Looks like it just checks whether `deprecatedThemeOptions` has some values, which is always true

Reply:
it should map deprecated options to themeVars

```
name: "brandTitle";
url: "brandUrl";
```

Reply:
I mean that
```
if (Object.values({name: 'brandTitle', url: 'brandUrl'}).find(v => !!v))
```

evaluates to
```
if (['brandTitle', 'brandUrl'].find(v => !!v))

// same as
if ('brandTitle')

// same as
if (true)
```